src/tools/sfdk/sdkmanager.cpp

```diff
@@ -310,6 +314,87 @@ void SdkManager::setEnableReversePathMapping(bool enable)
     s_instance->m_enableReversePathMapping = enable;
 }
 
+Device *SdkManager::configuredDevice(QString *errorMessage)
+{
+    Q_ASSERT(errorMessage);
+
+    const Option *const deviceOption = Dispatcher::option("device"); // FIXME
```

Comment:
Something you've forgotten or planning to fix in the near future?

Reply:
The use of the string literal here was a bit controversial to me.

I prefer NOT to use constants blindly for every primitive value. E.g. when a constant is used for a value that is specified by an external entity, where the context is clear and the value is not likely to change, all it adds is an unnecessary overhead for both the writer and readers of the code. Consider embedding an SQL query in C++ code - would we define a constant for every SQL keyword and build the queries using that? - not likely. A deterrent example is how do we work with vboxmanage :)

The case with "device" option is not that straightforward. It comes from sfdk module definition, which isn't really external to sfdk. It is only external to its C++ side. At the same time there is currently no mechanism to share definitions between those. So what the FIXME follows here is an opportunity to add such a mechanism and use it here.

Given that 1) I am not aware of any other case similar to this, 2) it is likely to fail soon when the other side changes and 3) improving it wouldn't be for free, I think relying on the assert below is just enough. I will remove the FIXME.

Reply:
Ok. Thanks for the explanation :)